pkg/cloud/services/compute/instance.go

```diff
@@ -206,10 +204,15 @@ func (s *Service) createInstance(eventObject runtime.Object, clusterName string,
 		})
 	}
 
-	if instanceSpec.Subnet != "" && accessIPv4 == "" {
-		if err := s.deletePorts(eventObject, portList); err != nil {
-			return nil, err
+	// Ensure we delete the ports we created on any error prior to instance create
+	deletePorts := portList
+	defer func() {
```

Comment:
I think this is cool syntax, but its also way more confusing then just having the function return an error and checking it when it returns.

Reply:
That's what the previous code did, and it missed several cases already 😉 And I'm adding another one.

This is more robust than requiring adding the same check at every potential exit point of the function. I will eventually add unit tests for this behaviour, btw.

Reply:
ok, I can live with it I suppose.

Reply:
I have now added unit tests!